### PR TITLE
Minor punctuation and stylistic Readme edits

### PR DESCRIPTION
**Pull request by mohamedAdhamc**
minor changes however I believe after this the punctuation should all be correct.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,14 +19,14 @@ Windows), mobile platforms (Android, iOS), as well as Web-based platforms
 
 ## Free, open source and community-driven
 
-Godot is completely free and open source under the very permissive [MIT license](https://godotengine.org/license).
+Godot is completely free, and open-source under the very permissive [MIT license](https://godotengine.org/license).
```
Comment (sairam4123) [inline README.md:22]: I think adding a comma before `and` doesn't feel grammatically correct for me.

File: README.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ To get started contributing to the project, see the [contributing guide](CONTRIB
 
 ## Documentation and demos
 
-The official documentation is hosted on [ReadTheDocs](https://docs.godotengine.org).
+The official documentation is hosted on [Read The Docs](https://docs.godotengine.org).
```
Comment (sairam4123) [inline README.md:60]: ReadTheDocs is a website, so I think it's better to leave it as it is.

**Review by sairam4123 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Calinou) [reply README.md:60]: The official spelling is ["Read the Docs"](https://readthedocs.org/) (case-sensitive).

File: README.md (hunk 1 above)
Comment (Calinou) [reply README.md:22]: I'd also keep that sentence as it is currently. Having too many commas can reduce readability.

File: README.md (hunk 2 above)
Comment (mohamedAdhamc) [reply README.md:60]: > The official spelling is ["Read the Docs"](https://readthedocs.org/) (case-sensitive), not "Read The Docs".

Ok thanks for the info will update the proposal.

**Comment by mohamedAdhamc** [comment]
@Calinou fixed the parts you guys mentioned. Thanks!

**Review by mhilbrunner — APPROVED** [review]

**Comment by akien-mga** [comment]
Sorry for the delay reviewing this.

I wasn't convinced that those changes are actually better, like usual in such stylistic edits. Style is a complex matter and to change things sourced in common usage, we need to either be very pedantic and go against the use, or find very clear international consensus on the style rule.

Here for open source, I don't think the change is wrong, but I don't think the previous language was wrong either. And "open source" without hyphen seems to actually be more common in the tech sphere.

Here's what I found:

- Google says *never to hyphenate* open source: https://opensource.google/documentation/reference/comms/style-guide
- Microsoft says to hyphenate open source only when used as an adjective preceding a noun ("open-source software"): https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/o/open-source. This is not the case in the two edits here.
- Wikipedia uses both "open-source" and "open source" throughout their article on "open-source software": https://en.wikipedia.org/wiki/Open-source_software. They partially seem to use the rule to hyphenate open source when used as an adjective preceding a noun, and not in other occurrences, but that's not done consistently.

The added comma doesn't seem to add much, I think both are fine, so again I'm not particularly interested in changing it if it wasn't wrong.

"ReadTheDocs" changing to "Read the Docs" is correct, but that's actually something that I'd prefer to rework more in depth eventually as it's not really relevant to the reader which platform is used to build the documentation. For most users this is not Read the Docs but Godot's own documentation website.

**Pull request closed without merging**